app/post_metadata.go

```diff
@@ -317,14 +317,24 @@ func getImagesInMessageAttachments(post *model.Post) []string {
 	return images
 }
 
-func (a *App) getLinkMetadata(requestURL string, useCache bool) (*opengraph.OpenGraph, *model.PostImage, error) {
+func (a *App) getLinkMetadata(requestURL string, timestamp int64, isNewPost bool) (*opengraph.OpenGraph, *model.PostImage, error) {
 	requestURL = resolveMetadataURL(requestURL, a.GetSiteURL())
 
+	timestamp = model.FloorToNearestHour(timestamp)
+	hash := model.GenerateLinkMetadataHash(requestURL, timestamp)
```

Comment:
For me is kinda weird to have 2 functions that get the data from cache and each one receive different parameters. Maybe we can receive in both cases the `requestURL` and the `timestamp` for the `getLinkMetadataFromCache` and `getLinkMetadataFromDatabase`. Actually I would move the FloorToNearstHour inside both functions. I know the specifications document propose to move it outside, but the quantity of duplicated code is low, and I think is clearer the separation of concerns of each function.

Reply:
Yeah, I'm thinking that might be a good idea. I was trying to avoid generating the hash multiple times, but that's probably a negligible performance difference to make it much more readable